assignments/this.js

```diff
@@ -13,14 +13,32 @@
 
 // code example for Window Binding
 
+console.log(this);
+
 // Principle 2
 
 // code example for Implicit Binding
 
+const MyObject = function(attr) {
+    this.name = attr.name;
+    this.favActivity = attr.favActivity;
+};
+MyObject.prototype.doStuff = function() {
+    console.log(`${this.name}'s favorite activity is ${this.favActivity}`);
+};
+
 // Principle 3
 
 // code example for New Binding
+let talking = function() {
+    console.log(`My name is ${this.name}`);
+}
+talking = talking.bind({name: 'Bob'});
+talking();
 
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
+const me = {name: 'Joe', favActivity: 'woodworking'};
+MyObject.prototype.doStuff.call(me, me.favActivity);
```

Comment:
I am very confused right here, not sure what is going on here I am not sure what context this would work in or if it would work. probably need to add on to this example to test this explicit binding.